plugins/hls-class-plugin/src/Ide/Plugin/Class.hs

```diff
@@ -234,18 +247,38 @@ codeAction state plId (CodeActionParams _ _ docId _ context) = liftIO $ fmap (fr
           _ -> panic "Ide.Plugin.Class.findClassFromIdentifier"
     findClassFromIdentifier _ (Left _) = panic "Ide.Plugin.Class.findClassIdentifier"
 
+    findImplementedMethods :: NormalizedFilePath -> Range -> MaybeT IO [T.Text]
+    findImplementedMethods docPath range = do
+        (HAR {hieAst = hf}, pmap) <- MaybeT . runAction "classplugin" state $ useWithStale GetHieAst docPath
+        pure
+            $ concat
+            $ pointCommand hf (fromJust (fromCurrentRange pmap range) ^. J.start & J.character -~ 1)
+            $ map (T.pack . getOccString) . rights . findInstanceValBindIdentifiers
+
+    findInstanceValBindIdentifiers :: HieAST a -> [Identifier]
```

Comment:
This could do with some haddock and/or explanation. what is it doing and why?

Reply:
I added an explanation, please let me know if it makes sense.

Reply:
Why can't we just transitively look for all the things that are `InstanceValBind`s underneath, rather than having to specifically say which things we'll recurse through? I'd have thought that the only `InstanceValBind`s in a instance declaration would be precisely the already-implemented methods?

Reply:
Hmm, yeah makes the code much simpler: 

```haskell
    findInstanceValBindIdentifiers :: HieAST a -> [Identifier]
    findInstanceValBindIdentifiers ast =
        let valBindIds = Map.keys
                         . Map.filter (any isInstanceValBind . identInfo)
                         $ getNodeIds ast
        in valBindIds <> concatMap findInstanceValBindIdentifiers (nodeChildren ast)
```

I guess this will perform worse as this is now traversing more of the AST, but maybe this is not as bad?

Reply:
Actually, this also makes the code compatible with older GHCs, so I am going to do commit it.